synthnn/core/musical_extensions.py

```
@@ -156,16 +156,23 @@ def analyze_and_retune(self, foreign_signal: np.ndarray,
         
         return detected_mode
     
-    def _retune_to_mode(self, new_base_freq: float, mode_intervals: List[float]) -> None:
+    def _retune_to_mode(self, new_base_freq: float, mode_intervals) -> None:
```

Ruff (0.11.9)

Missing type annotation for function argument `mode_intervals`

(ANN001)